docs/Migration Guide.md

```diff
@@ -5,11 +5,78 @@ of SpiderMonkey to the next ESR version.
 
 ## ESR 102 to ESR 115 ##
 
+### More Versatile `JS_InitClass` ###
+
+In previous versions of SpiderMonkey, the name of the constructor
+function added by `JS_InitClass` was always the name of the prototype
+object's `JSClass`.
+This limitation caused the instances and the prototype to always share
+the same `JSClass` and `JSClassOps` specification.
+If the prototype should behave differently than the instances, custom
+checks were needed.
+Now `JS_InitClass` accepts a `const char *` parameter for naming the
+constructor function, so a different `JSClass *` can be used for the
+prototype object.
+If `nullptr` is passed, the prototype object will be a plain JS object.
+
+### Module Import ###
+
+Previously, `JS::FinishDynamicModuleImport()` when passed an
+`evaluationPromise` argument that was null, would take the pending
+exception from the context and reject the import promise with it.
+This is no longer the case.
```

Comment:
I haven't found the underlying change for this behavior change.
I'll look into this tomorrow.

Reply:
I actually cannot reproduce it with a minimal example. So maybe I misunderstood the behaviour change. In case it helps figure out what was going on, here's the change I had to make: https://gitlab.gnome.org/GNOME/gjs/-/commit/572073dff01c1cf199c08a4e537f13ad5f99a290

Maybe I was just originally misusing the API :shrug: If we can't figure it out, I can drop this section from the migration guide.

Reply:
tested with mozjs 115.1.0, and the test added by the patch passes even without the patch for `gjs/module.cpp` file applied.
is there more context about what went wrong with unpatched code?

Reply:
Hmm! At the time, I was getting an exception without a message, instead of the exception thrown by the module import, so the test would fail. I wonder if it was just a regression in the 115.0 alpha which I was testing with. Anyway, it sounds like I should drop this section. Thanks for looking into it!